src/main/java/org/skriptlang/skript/lang/properties/Property.java

```diff
@@ -0,0 +1,255 @@
+
+	/**
+	 * Creates a new property. Prefer {@link #of(String, String, String, SkriptAddon, Class)}.
+	 *
+	 * @param name the name of the property
+	 * @param provider the addon that provides this property
+	 * @param handler the handler class for this property
+	 * @see #of(String, String, String, SkriptAddon, Class)
+	 */
+	public Property(String name, String description, String[] since, SkriptAddon provider, @NotNull Class<? extends Handler> handler) {
+		this.name = name.toLowerCase(Locale.ENGLISH);
+		this.description = description;
+		this.since = since;
+		this.provider = provider;
+		this.handler = handler;
+	}
+
+	/**
+	 * Gets a documentation-friendly ID for this property, based on its name.
+	 * May be overridden to provide a custom ID.
+	 *
+	 * @return a documentation-friendly ID for this property
+	 */
+	public String getDocumentationID() {
+		return name.replace(' ', '-').toLowerCase(Locale.ENGLISH);
+	}
+
+	/**
+	 * Helpful registration shortcut.
+	 * @return whether the property was successfully registered.
+	 */
+	public boolean register() {
+		return PROPERTY_REGISTRY.register(this);
+	}
+
+	/**
+	 * Creates a new property.
+	 *
+	 * @param name the name of the property
+	 * @param description a brief description of the property
+	 * @param provider the addon that provides this property
+	 * @param handler the handler class for this property
+	 * @param <HandlerClass> the type of the handler class
+	 * @param <Handler> the type of the handler
+	 * @return a new property
+	 */
+	public static <HandlerClass extends PropertyHandler<?>, Handler extends HandlerClass> Property<Handler> of(
+			@NotNull String name,
+			@NotNull String description,
+			@NotNull String since,
+			@NotNull SkriptAddon provider,
+			@NotNull Class<HandlerClass> handler) {
+		//noinspection unchecked
+		return (Property<Handler>) new Property<>(name, description, new String[]{since}, provider, handler);
+	}
+
+	/**
+	 * Creates a new property.
+	 *
+	 * @param name the name of the property
+	 * @param description a brief description of the property
+	 * @param since the version[s] that this property was added/updated.
+	 * @param provider the addon that provides this property
+	 * @param handler the handler class for this property
+	 * @param <HandlerClass> the type of the handler class
+	 * @param <Handler> the type of the handler
+	 * @return a new property
+	 */
+	public static <HandlerClass extends PropertyHandler<?>, Handler extends HandlerClass> Property<Handler> of(
+		@NotNull String name,
+		@NotNull String description,
+		@NotNull String @NotNull [] since,
+		@NotNull SkriptAddon provider,
+		@NotNull Class<HandlerClass> handler) {
+		//noinspection unchecked
+		return (Property<Handler>) new Property<>(name, description, since, provider, handler);
+	}
+
+	/* ****************************************************
+	 * DEFAULT PROPERTIES
```

Comment:
I might prefer to see this elsewhere. What do you think about a `properties` module in `common`? You can move all of the `PropExpr` syntax within it.

Reply:
I specifically had it here so users can do `Property.` and see all of Skript's properties. I do get the point about it getting messy over time, but I like the convenience. Maybe it can just be an intermediate